applications/new_schema.py

```diff
@@ -0,0 +1,81 @@
+import django_filters
+import graphene
+from graphene_django.filter import DjangoFilterConnectionField
+from graphene_django.types import DjangoObjectType
+from graphql_jwt.decorators import login_required, superuser_required
+
+from .models import BerthApplication, BerthSwitch, HarborChoice
+from .schema import BerthSwitchType as OldBerthSwitchType
+
+
+class HarborChoiceType(DjangoObjectType):
+    harbor = graphene.String()
+
+    class Meta:
+        model = HarborChoice
+        exclude = ("id", "application")
+
+    def resolve_harbor(self, info, **kwargs):
+        if self.harbor:
+            return self.harbor.servicemap_id
+        return None
+
+
+class BerthSwitchType(OldBerthSwitchType):
+    harbor = graphene.String()
+
+    class Meta:
+        model = BerthSwitch
+
+    def resolve_harbor(self, info, **kwargs):
+        if self.harbor:
+            return self.harbor.servicemap_id
+        return None
+
+
+class BerthApplicationFilter(django_filters.FilterSet):
+    switch_applications = django_filters.BooleanFilter(
+        field_name="berth_switch", method="filter_berth_switch"
+    )
+
+    def filter_berth_switch(self, queryset, name, value):
+        lookup = "__".join([name, "isnull"])
+        return queryset.filter(**{lookup: not value})
+
+
+class BerthApplicationTypeNode(DjangoObjectType):
```

Comment:
this can be called just `BerthApplicationNode` 🙂

Reply:
the convention we "kinda" adopted:
- regular GQL type = `<ModelName>Type`
- type with node interface = `<ModelName>Node`
any class names that do not follow this convention can be renamed at any conveneint time 🙂